### PR TITLE
mgr/dashboard: Prevent API call on every keystroke

### PR DESCRIPTION
**Pull request by votdev**
Every keystroke for the username in the RGW user form triggers an API call

Fixes: https://tracker.ceph.com/issues/25161

Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts [hunk 1]
```diff
@@ -274,25 +280,31 @@ export class RgwUserFormComponent implements OnInit {
 
   /**
    * Validate the username.
+   * @param {number|Date} dueTime The delay time to wait before the
+   *   API call is executed. This is useful to prevent API calls on
+   *   every keystroke. Defaults to 500.
    */
-  userIdValidator(): AsyncValidatorFn {
+  userIdValidator(dueTime = 500): AsyncValidatorFn {
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: This doesn't solves the problem in my opinion as it's only delaying the problem. It reduces it a bit, but the problem is still there. Why not validate on focus loss and before submitting?

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: This is incorrect, if there are keystrokes during this 500ms, then every delayed request before is canceled. If there are no more keystrokes during this time, the last value is submitted via REST call.

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: > Why not validate on focus loss and before submitting?

I want to have feedback as soon as possible during typing. We have a modern UI that supports this, so why not using it.

Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts]: How about firing after 750ms and focus loss? Than you depend on which is fired first.

**Pull request merged**